library/src/main/kotlin/com/liferay/vulcan/consumer/screens/ThingScreenlet.kt

```diff
@@ -3,83 +3,114 @@ package com.liferay.vulcan.consumer.screens
 import com.liferay.vulcan.consumer.screens.views.BaseView
+import com.liferay.vulcan.consumer.screens.views.Custom
+import com.liferay.vulcan.consumer.screens.views.Detail
+import com.liferay.vulcan.consumer.screens.views.Row
+import com.liferay.vulcan.consumer.screens.views.Scenario
 import okhttp3.HttpUrl
 
 open class BaseScreenlet @JvmOverloads constructor(
-    context: Context, attrs: AttributeSet?, defStyleAttr: Int = 0, defStyleRes: Int = 0) :
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0, defStyleRes: Int = 0) :
     FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
 
     var layout: View? = null
 }
 
 class ThingScreenlet @JvmOverloads constructor(
-    context: Context, attrs: AttributeSet?, defStyleAttr: Int = 0, defStyleRes: Int = 0) :
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0, defStyleRes: Int = 0) :
     BaseScreenlet(context, attrs, defStyleAttr, defStyleRes) {
 
     var screenletEvents: ScreenletEvents? = null
 
-    val layoutIds: Map<String, Int> = mapOf(
-        "BlogPosting" to R.layout.blog_posting_default,
-        "Collection" to R.layout.collection_default
-    )
+    var scenario: Scenario = Detail
 
-    val reference: Int
+    companion object {
+        val layoutIds: MutableMap<String, MutableMap<Scenario, Int>> = mutableMapOf(
+            "BlogPosting" to BlogPosting.DEFAULT_VIEWS,
+            "Collection" to Collection.DEFAULT_VIEWS,
+            "Person" to Person.DEFAULT_VIEWS
+        )
+    }
+
+    val layoutId: Int
+
+    var thing: Thing? by observe {
+        val layoutId = getLayoutIdFor(thing = it) ?: R.layout.thing_default
 
-    var thing: Thing? = null
-        set(value) {
-            field = value
-            viewModel?.thing = value
+        layout?.also {
+            this.removeView(it)
         }
 
-    val viewModel: ViewModel? get() = layout as ViewModel
+        layout = this.inflate(layoutId)
 
-    fun load(thingId: String, onComplete: ((ThingScreenlet) -> Unit)? = null) {
-        fetch(HttpUrl.parse(thingId)!!) {
-            val layoutId = if (reference == 0)
-                it.component1()?.type?.get(0)
-                    ?.let { layoutIds[it] }
-                    ?: R.layout.thing_default
-            else reference
+        addView(layout)
 
-            layout = this.inflate(layoutId)
+        baseView?.apply {
+            screenlet = this@ThingScreenlet
+            thing = it
+        }
+    }
 
-            addView(layout)
+    val baseView: BaseView? get() = layout as? BaseView
 
-            (layout as? BaseView)?.screenlet = this
+    fun load(thingId: String, scenario: Scenario? = null, onComplete: ((ThingScreenlet) -> Unit)? = null) {
+        fetch(HttpUrl.parse(thingId)!!) {
+            if (scenario != null) {
+                this.scenario = scenario
+            }
 
-            it.failure { viewModel?.showError(it.message) }
+            thing = it.component1()
```

Comment:
They are standard for destructuring in Kotlin: https://kotlinlang.org/docs/reference/multi-declarations.html